src/api/app/models/kiwi/image/xml_builder.rb

```diff
@@ -14,12 +14,56 @@ def build
         doc = update_packages(doc)
         doc = update_repositories(doc)
         doc = update_description(doc)
+        doc = update_preferences(doc)
 
         Nokogiri::XML(doc.to_xml, &:noblanks).to_xml(indent: kiwi_indentation(doc))
       end
 
       private
 
+      def update_preferences(document)
+        return document if @image.preference_type.blank?
+
+        # <preferences> block does not exist
+        if document.xpath('image/preferences').any?
```

Comment:
Do we need to check this? A kiwi file without preferences is anyway invalid! I understand that it seems that it could happen that there are kiwi files without a preference section but then we should avoid importing it, no?

Reply:
It's not only about importing. It's currently possible to remove the preferences from a kiwi file, via the plain text editor.

Reply:
I  just realized you meant the importing / parsing of the file via the kiwi editor. I agree it would be nicer to catch this case by validating the file during the import.
Then we don't need all these checks.

So if we already have an xml validation build in, I agree with @ChrisBr to drop the additional check. Otherwise we need to keep it for now.